### ISSUE TITLE
Tasks get reassigned if the place of stay is changed

### ISSUE THREAD

**Issue by kwa20** [hzi-braunschweig/SORMAS-Project#6631]
### Bug Description

Tasks get reassigned to the responsible user when the place of stay changes. This includes changing region or district as well as changing the health facility (includes home). When changing the facility, selecting 'edit data' or 'transfer case' does not make a difference. Both lead to reassignment of tasks.

If there is no responsible officer in the respective jurisdiction, the tasks of that case will get randomly reassigned to any user that has access to that case.

This is quite problematic because users cannot properly track the tasks they're supposed to fulfill if the assignee keeps changing randomly without them knowing. 
### Steps to Reproduce

1. Create a case and create several tasks with all having different assignee (These should differ from the responsible user of that case)
2. Change the place of stay in the following ways
    a. change the facility and select 'edit data' or 'transfer case' (try both separately)
    b. select 'place of stay differs from the jurisdiction' and select a differing region/district
3. Check the assignee of the cases' tasks
4. Repeat the process with a case that has no available responsible user to be set

### Expected Behavior

Tasks should never get randomly reassigned without the user knowing, especially, if the users assigned to the task still have access to that case as before.
### Screenshots

Before:
<img width="206" alt="taskAssignment" src="https://user-images.githubusercontent.com/70317594/132215096-f325a719-37b6-41a1-826e-48e7b871d4c7.PNG">
<img width="393" alt="taskAssignment3" src="https://user-images.githubusercontent.com/70317594/132215092-251b96fd-1ef3-4d4a-bdb2-47c053b576f2.PNG">
<img width="198" alt="taskAssignment2" src="https://user-images.githubusercontent.com/70317594/132215099-d675b329-03ea-47c1-b790-2f18909ba063.PNG">

Change:
<img width="392" alt="taskAssignment5" src="https://user-images.githubusercontent.com/70317594/132215095-2a21bb01-46b5-49d9-bab3-9ab96a1e0d79.PNG">

After:
<img width="208" alt="taskAssignment4" src="https://user-images.githubusercontent.com/70317594/132215094-028264d3-cf3b-4161-ab6b-f9a6b189f9ba.PNG">
### System Details

* Device:
* SORMAS version: 1.63.0
* Android version/Browser: firefox
* Server URL: test1401.sormas-oegd.de
* User Role: admin, national user

### Additional Information

cf. https://github.com/hzi-braunschweig/SORMAS-Project/issues/4507 as a similar previous problem

**Comment by Marko-ilmkreis** [comment]
@kwa20 Thanks for the issue. I was about about to open the same as I get aware of these very annoying bug.

**Comment by kwa20** [comment]
@Marko-ilmkreis You're very welcome! If you have any further information that are not covered by the description, feel free to add them

**Comment by MateStrysewske** [comment]
Task: Only re-assign tasks when the jurisdictions of a case changes if the currently assigned user does not have access to the task anymore (i.e. is not among the list of potential task assignees).

**Comment by syntakker** [comment]
There are test failures in the android tests (#6714) and an incosistency of the implementations of Vaadin and Android app (#6715).
Both issues are beyond the scope of this issue and therefore treated separately.

**Comment by kwa20** [comment]
Do we have a milestone for this issue if it's already under review?

**Comment by MateStrysewske** [comment]
@kwa20 Only once it's merged into the development branch

**Comment by roxanamlendea** [comment]
Validated ticket on the latest version of Sormas 1.65.0-SNAPSHOT on environment test-de.sormas.netzlink.com
